src/data-loading.test.tsx

```diff
@@ -137,6 +137,90 @@ it('cap wrap data load function', async () => {
     })
 })
 
+it('component can provide additional arguments dynamically', async () => {
+    const resources = new DataLoaderResources<{}>()
+    const { middleware, createRegisterableComponentWithData } = init<{}>(resources)
+
+    const lengthCalculatorWithMultiplierDataDefinition: DataDefinition<
+        { dataArg: string },
+        number,
+        {},
+        { multiplier: number }
+    > = {
+        // Additional params can come from anywhere, for instance redux or
+        // other environmental variables (window.location?)
+        getAdditionalParams: () => {
```

Comment:
I'm a bit confused about why you would want to mix these into the `props` argument if you know the values already at this point and can just use them directly in loadData? Why have a hook that bakes this into the library?

Reply:
Because the props are used as the cache key in the data loader. 

The load function will not be called again because the params are the same.

Reply:
So if you were pass window.location or redux state into `getAdditionalParams` It would reload given the case were thse things changed at some point?